pkg/status/status.go

```diff
@@ -85,10 +87,45 @@ func GetStatus() (map[string]interface{}, error) {
 	}
 
 	stats["systemProbeStats"] = getSystemProbeStats()
+	stats["apmStats"] = GetAPMStatus()
 
 	return stats, nil
 }
 
+// GetAPMStatus returns a set of key/value pairs summarizing the status of the trace-agent.
+// A key "outcome" is always returned and it may be either "error" or "success".
+func GetAPMStatus() map[string]interface{} {
```

Comment:
I think this should be done by the `cmd` and not the agent. The command would query all agents to gather the needed info and then render it instead or relying on the core-agent.

Reply:
That's how I had it initially, but you will notice that this function is called from two places (it needs to be in a shared package):
* The `agent status` command
* The GUI code

This seems like the appropriate place to have it. I'm not sure why you say "relying on core-agent" since this is a shared package and doesn't necessarily by definition belong to the core-agent since we're all sharing the same repository now.

Does this change your perspective? If not, what better place to have this shared function do you see?

Reply:
What I meant was: I don't think the command line should call the core-agent which in turn call the trace-agent. Unless I'm missing something, this code is running in the core-agent process, no ?

For me the command line should call the core-agent to get his status and then the expvar from the trace-agent to get his, and then merge both and render the page. I don't want the core-agent to be a router toward other agent's processes.

Reply:
Sorry but I'm still confused here. It feels to me that what you're describing is exactly what's happening: the `agent status` command reveals the status for all agents now (except processes).

This is what we're trying to fix with this PR as part of another discussion where the `agent status` was not revealing the status for APM. For customers the concept of "multiple agents" does not exist and we can not ask them to do both `agent status` and then `trace-agent -info` (which seems really weird) as it gets tedious for both them and support. `agent status` should just return everything as the main entry point.